### PR TITLE
Feature 2233/give check ins theming powers

### PR DESCRIPTION
**Pull request by jackkeller**
- adds _variables.css file to control theming natively
- overwrites a few things in App.jsx where native css variables were not doing the trick

### CODE REVIEW

File: web-ui/.prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_variables.css
```
Comment (mvolkmann) [inline web-ui/.prettierignore:1]: What does Prettier do to this that is bad?

File: web-ui/src/App.css [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+@import 'styles/_variables.css';
+
+.App {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+
+  a,
+  a:visited {
+    color: var(--checkins-palette-primary-link);
```
Comment (mvolkmann) [inline web-ui/src/App.css:10]: What do you think about replacing "checkins-palette" with just "checkins" everywhere just to shorten the names?

File: web-ui/src/App.jsx [hunk 3]
```diff
@@ -10,66 +10,79 @@ import { AppContextProvider } from './context/AppContext';
 import SnackBarWithContext from './components/snackbar/SnackBarWithContext';
 import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
 import { LocalizationProvider } from '@mui/x-date-pickers';
-import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+import {
+  experimental_extendTheme as extendTheme,
```
Comment (mvolkmann) [inline web-ui/src/App.jsx:15]: What is the significance of starting names with "experimental"?

File: web-ui/src/App.jsx [hunk 4]
```diff
@@ -10,66 +10,79 @@ import { AppContextProvider } from './context/AppContext';
...
+  Experimental_CssVarsProvider as CssVarsProvider
+} from '@mui/material/styles';
 
 import './App.css';
 
-const customHistory = createBrowserHistory();
+function getUserColorScheme() {
+  console.log('getting your scheme');
```
Comment (mvolkmann) [inline web-ui/src/App.jsx:22]: I guess you'll remove this when finished testing.

File: web-ui/src/App.jsx [hunk 5]
```diff
@@ -10,66 +10,79 @@ import { AppContextProvider } from './context/AppContext';
...
+  if (
+    window.matchMedia &&
+    window.matchMedia('(prefers-color-scheme: dark)').matches
```
Comment (mvolkmann) [inline web-ui/src/App.jsx:25]: Does it work to change this to `window?.matchMedia(...` and remove line 24?
Same on line 30.

File: web-ui/src/App.jsx [hunk 6]
```diff
@@ -10,66 +10,79 @@ import { AppContextProvider } from './context/AppContext';
...
-    secondary: {
-      light: '#ffe8a2',
-      main: '#feb672',
-      dark: '#c88645',
-      contrastText: '#000'
+const theme = extendTheme({
+  cssVarPrefix: 'checkins',
+  colorSchemes: {
+    light: {
+      palette: {
```
Comment (mvolkmann) [inline web-ui/src/App.jsx:42]: Do `light` and `dark` ever contain properties other than `palette`?
I'm wondering if that layer of the hierarchy can be removed.

File: web-ui/src/App.jsx [hunk 7]
```diff
@@ -10,66 +10,79 @@ import { AppContextProvider } from './context/AppContext';
...
-    },
-    MuiTextField: {
-      defaultProps: {
-        variant: 'standard'
-      }
     }
   }
 });
 
+const customHistory = createBrowserHistory();
```
Comment (mvolkmann) [inline web-ui/src/App.jsx:72]: Is this used?

File: web-ui/src/App.css (hunk 2 above)
Comment (jackkeller) [reply web-ui/src/App.css:10]: Palette is something mui adds.

File: web-ui/.prettierignore (hunk 1 above)
Comment (jackkeller) [reply web-ui/.prettierignore:1]: It breaks up the color-mix variables, not dire but I made a preference call.

File: web-ui/src/App.jsx (hunk 3 above)
Comment (jackkeller) [reply web-ui/src/App.jsx:15]: It was just what the docs said to use these features.

File: web-ui/src/App.jsx (hunk 4 above)
Comment (jackkeller) [reply web-ui/src/App.jsx:22]: Yes.

**Comment by jackkeller** [comment]
I'm going to probably try a cherry pick branch to remove the merge commits then a new PR.

File: web-ui/src/components/received_request_card/ReceivedRequestCard.jsx [hunk 8]
```diff
@@ -21,21 +22,21 @@ const classes = {
...
-  [`& .${classes.greenTypography}`]: {
-    color: '#006400'
-  },
-  [`& .${classes.darkGrayTypography}`]: {
-    color: '#333333'
-  },
-  [`& .${classes.grayTypography}`]: {
-    color: 'gray'
-  }
+  // [`& .${classes.redTypography}`]: {
```
Comment (mvolkmann) [inline web-ui/src/components/received_request_card/ReceivedRequestCard.jsx:25]: Can these commented lines be deleted?

File: web-ui/src/styles/_variables.css [hunk 9]
```diff
@@ -0,0 +1,100 @@
+:root {
```
Comment (mvolkmann) [inline web-ui/src/styles/_variables.css:1]: What's the reason this filename starts with an underscore?

File: web-ui/src/styles/_variables.css [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+  --primary-link-light: var(--primary-light);
+  --primary-link-light-hover: #6084d1;
+  --primary-link-dark: color-mix(in oklab, var(--oci-dark-blue), var(--white) 35%);
+  --primary-link-dark-hover: #4f5f8c;
+  --action: color-mix(in oklab, var(--black), var(--white) 50%);
+  --action-disabled: color-mix(in oklab, var(--black), var(--white) 5%);
+
+  --checkins-palette-action-disabledOpacity: 0.58;
+
+  /* MUI Prefixes */
```
Comment (mvolkmann) [inline web-ui/src/styles/_variables.css:38]: What is the connection between these variables and MUI? Are they only used to style MUI components?

**Comment by jackkeller** [comment]
Closing with new PR that has less inherited commits from develop

**Pull request closed without merging**